### PR TITLE
Log NodeTemplate updates

### LINKED ISSUE #292: [BUG] ALTs are not synced properly to nodes
**Issue:** Custom annotations/labels/taints defined on the shoot yaml are sometimes missed on the newly created nodes

**Observations made:** The nodes went up to 3 once, all missing the label some minutes. The label seemed to be present for the first few minutes [at least for one of them], then the label disappeared

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**:
A minor change of logging of NodeTemplate updates.

**Which issue(s) this PR fixes**:
Fixes # 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
```

### CODE REVIEW

File: pkg/controller/deployment_sync.go [hunk 1]
```diff
@@ -275,13 +275,7 @@ func (dc *controller) getNewMachineSet(d *v1alpha1.MachineDeployment, isList, ol
...
 		if needsUpdate {
 			var err error
-			newStatus := d.Status
-			if d, err = dc.controlMachineClient.MachineDeployments(d.Namespace).Update(d); err != nil {
-				return nil, err
-			}
-			dCopy := d.DeepCopy()
-			dCopy.Status = newStatus
-			if d, err = dc.controlMachineClient.MachineDeployments(dCopy.Namespace).UpdateStatus(dCopy); err != nil {
+			if d, err = dc.controlMachineClient.MachineDeployments(d.Namespace).UpdateStatus(d); err != nil {
```
Comment (hardikdr) [inline pkg/controller/deployment_sync.go]: ref: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/sync.go#L175

**Comment by hardikdr** [comment]
Can't think of a good test case to be added, it only removes the [mostly]un-used update call and adds logging. suggestions?

File: pkg/controller/deployment_util.go [hunk 2]
```diff
@@ -435,6 +435,10 @@ func copyMachineDeploymentNodeTemplatesToMachineSet(deployment *v1alpha1.Machine
 
 	isNodeTemplateChanged := !(apiequality.Semantic.DeepEqual(deployment.Spec.Template.Spec.NodeTemplateSpec, is.Spec.Template.Spec.NodeTemplateSpec))
 
+	if isNodeTemplateChanged {
+		glog.V(2).Infof("Observed a change in NodeTemplate of Machine Deployment %s. Changing NodeTemplate from %+v to \n %+v.", deployment.Name, is.Spec.Template.Spec.NodeTemplateSpec, deployment.Spec.Template.Spec.NodeTemplateSpec)
+	}
+
 	if isNodeTemplateChanged {
```
Comment (swapnilgm) [inline pkg/controller/deployment_util.go]: Why do you have duplicate `if` checks on line 438 and line 442 consecutively?

Comment (hardikdr) [reply pkg/controller/deployment_util.go]: oops, that's a mistake, thanks .. .. 👍

**Pull request merged**